### DISCUSSION TITLE
Audio low volume

### DISCUSSION THREAD

**Discussion by bzeb** [google/oboe#1707] [Q&A]
Within the PJSIP stack is an Android audio oboe implementation. On several terminals, I noticed a low volume on the speaker or in the handset. I have the signal amplification setting on the PJSIP at maximum. It's a little better but still too quiet. There are no such problems when using OpenSLES.

**Comment by jamessavery** [comment]
I've found across the board, Android produces lower tones compared to iOS

Also, Oboe is just a wrapper for OpenSLES - https://developer.android.com/ndk/guides/audio/opensl. Theoretically, there shouldn't be a difference at all

**Comment by philburk** [comment]
Is PJSIP from https://www.pjsip.org?

I have also seen variations in volume between different data paths through the DSP.
This is controlled by the vendor software. We may need to require consistency across data paths.

> Oboe is just a wrapper for OpenSLES

It is also a wrapper for AAudio. And AAudio can have two data paths, MMAP and Legacy.
So there are three possible data paths when using Oboe.
Oboe does not change the volume. That happens below Oboe.
